mamba/mamba/api.py

```diff
@@ -0,0 +1,168 @@
+
+            cp = subdir.cache_path()
+            if cp.endswith(".solv"):
+                os.remove(subdir.cache_path())
+                cp = cp.replace(".solv", ".json")
+
+            self.local_repos[channelstr] = libmambapy.Repo(
+                self.pool, channelstr, cp, channelurl
+            )
+
+            self.local_repos[channelstr].set_priority(start_prio, 0)
+            start_prio -= 1
+
+    def solve(self, specs, pkg_cache_path=None):
+        """Solve given a set of specs.
+        Parameters
+        ----------
+        specs : list of str
+            A list of package specs. You can use `conda.models.match_spec.MatchSpec`
+            to get them to the right form by calling
+            `MatchSpec(mypec).conda_build_form()`
+        Returns
+        -------
+        transaction : libmambapy.Transaction
+            The mamba transaction.
+        Raises
+        ------
+        RuntimeError :
+            If the solver did not find a solution.
+        """
+        solver_options = [(libmambapy.SOLVER_FLAG_ALLOW_DOWNGRADE, 1)]
+        api_solver = libmambapy.Solver(self.pool, solver_options)
+        _specs = specs
+
+        api_solver.add_jobs(_specs, libmambapy.SOLVER_INSTALL)
+        success = api_solver.solve()
+
+        if not success:
+            error_string = "Mamba failed to solve:\n"
+            for s in _specs:
+                error_string += f" - {s}\n"
+            error_string += "\nwith channels:\n"
+            for c in self.channels:
+                error_string += f" - {c}\n"
+            pstring = api_solver.problems_to_str()
+
+            pstring = "\n".join(["- " + el for el in pstring.split("\n")])
+            error_string += f"\nThe reported errors are:\n{pstring}"
+            print(error_string)
+            raise RuntimeError("Solver could not find solution." + error_string)
+
+        if pkg_cache_path is None:
+            # use values from conda
+            pkg_cache_path = self.context.pkgs_dirs
+
+        package_cache = libmambapy.MultiPackageCache(pkg_cache_path)
+        return libmambapy.Transaction(api_solver, package_cache)
+
+
+def install(
+    env_name: str, specs: tuple = (), channels: tuple = (), target_platform: str = None
+):
+    """Install packages in a given environment.
+
+    Arguments
+    ---------
+    env_name : str
+        The name of the environment where to install the packages.
+    specs : tuple of str
+        The list of spec strings e.g. ['xeus-python', 'matplotlib=3'].
+    channels : tuple of str
+        The channels from which to pull packages e.g. ['default', 'conda-forge'].
+    Raises
+    ------
+    RuntimeError :
+        If the solver did not find a solution or if the installation failed.
+    """
+    prefix = pathlib.Path(os.environ["CONDA_PREFIX"]) / "envs" / env_name
+    (prefix / "conda-meta").mkdir(parents=True, exist_ok=True)
+    (prefix / "pkgs").mkdir(parents=True, exist_ok=True)
```

Comment:
Ideally we re-use the one from the "base" prefix here.

Reply:
So something like `os.environ["CONDA_PREFIX"]) / "pkgs"`?

Reply:
Yeah, unfortunately it's not that easy to find the base prefix. 

With micromamba, it's the `$MAMBA_ROOT_PREFIX` environment variable.

With conda it's the `import sys; sys.prefix`, but that assumes that conda is running from the base environment.

Reply:
Yeah I was under the impression it wouldn't be easy.

So you're suggesting like:
```python
base_prefix = None
if "MAMBA_ROOT_PREFIX" in os.environ:
    base_prefix = os.environ["MAMBA_ROOT_PREFIX"]
else:
    import sys
    base_prefix = sys.prefix
prefix = base_prefix / "envs" / env_name
(prefix / "conda-meta").mkdir(parents=True, exist_ok=True)
(base_prefix / "pkgs").mkdir(parents=True, exist_ok=True)
```
?

Reply:
yeah, and maybe make it override-able from a function argument

Reply:
thanks!